pyrsistent/_pclass.py

```diff
@@ -108,14 +108,17 @@ def set(self, *args, **kwargs):
         return self.__class__(_factory_fields=factory_fields, **kwargs)
 
     @classmethod
-    def create(cls, kwargs, _factory_fields=None):
+    def create(cls, kwargs, _factory_fields=None, ignore_extra=False):
         """
         Factory method. Will create a new PClass of the current type and assign the values
         specified in kwargs.
         """
         if isinstance(kwargs, cls):
             return kwargs
 
+        if ignore_extra:
+            kwargs = {k: v for k, v in kwargs.items() if k in cls._pclass_fields}
```

Comment:
Nitpick: I would probably invert this loop to avoid looping over a bunch of elements that are not relevant. In cases where  only a small subset of the passed in fields are actually used this could make a difference in peformance.

Eg.:
`{k: kwargs[k] for k in cls._pclass_fields if k in kwargs}`